update-changelog.sh

```diff
@@ -9,33 +9,86 @@ EMQX_VERSION=$2
 
 process_changes() {
     edition=$1
-    if ! grep -q "$EMQX_VERSION" en_US/changes/changes-$edition-v5.md; then
-        gsed -i "3i # $EMQX_VERSION\n" en_US/changes/changes-$edition-v5.md
-        gsed -i "5i ## Enhancements\n" en_US/changes/changes-$edition-v5.md
-        gsed -i "7i ## Bug Fixes\n" en_US/changes/changes-$edition-v5.md
+
+    [ ! -d $DEV_CHANGES_DIR/$edition ] && return
+
+    # Add a section for the new version if it doesn't exist along with the Enhancements and Bug Fixes sections
+    if ! grep -q "^## $EMQX_VERSION" en_US/changes/changes-$edition-v5.md; then
+        sed -i "3i ## $EMQX_VERSION\n" en_US/changes/changes-$edition-v5.md
+        sed -i "5i ### Enhancements\n" en_US/changes/changes-$edition-v5.md
+        sed -i "7i ### Bug Fixes\n" en_US/changes/changes-$edition-v5.md
     fi
 
-    enhancements_ln=$(grep -n Enhancements en_US/changes/changes-$edition-v5.md | head -n 1 | cut -d: -f1)
+    # Get the line number of the Enhancements section
+    enhancements_ln=$(grep -n '^### Enhancements' en_US/changes/changes-$edition-v5.md | head -n 1 | cut -d: -f1)
+    # Increment the line number to leave a blank line before the first enhancement
     enhancements_ln=$((enhancements_ln + 1))
 
     shopt -s nullglob
 
     for f in $DEV_CHANGES_DIR/$edition/fix-*.md $DEV_CHANGES_DIR/$edition/feat-*.md; do
-        pr_num="$(echo "${f}" | gsed -E 's/.*-([0-9]+)\.[a-z]+\.md$/\1/')"
-        if ! grep -q $pr_num en_US/changes/changes-$edition-v5.md; then
+        pr_num="$(echo "${f}" | sed -E 's/.*-([0-9]+)\.[a-z]+\.md$/\1/')"
+        if ! grep -q "^- \[#$pr_num\]" en_US/changes/changes-$edition-v5.md; then
+            if [ $pr_num -lt 10000 ]; then
```

Comment:
ok, this will buy some years time. ;)

Reply:
by then we should move everything back to the original repo 🤞